pcs/lib/corosync/node.py

```diff
@@ -39,46 +38,64 @@ def to_dto(self) -> CorosyncNodeAddressDto:
         )
 
 
-# TODO: add pcs.common.interface.dto.ImplementsToDto inheritance
-class CorosyncNode(namedtuple("CorosyncNode", "name addrs nodeid")):
+@dataclass(frozen=True)
+class CorosyncNode(ImplementsToDto):
     """A node loaded from corosync.conf"""
 
-    _addrs_plain = None
+    name: Optional[str]
+    addrs: Sequence[CorosyncNodeAddress]
+    nodeid: Optional[str]
 
-    def __new__(cls, name, addrs, nodeid):
-        return super().__new__(cls, name, tuple(addrs), nodeid)
+    def __post_init__(self) -> None:
+        # Make sure addrs is a tuple so it is not possible to change its items.
+        # Since the dataclass is frozen, object__setattr__ must be used.
+        object.__setattr__(self, "addrs", tuple(self.addrs))
```

Comment:
Is this really necessary? `Sequence` is immutable so mypy should make sure it's not being changed.

Reply:
Right, `mypy` prevents modifying `Sequence` here. Let's rely on that.